src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt

```diff
@@ -18,48 +18,45 @@ package io.github.gradlenexus.publishplugin
 
 import org.gradle.api.Action
 import org.gradle.api.NamedDomainObjectFactory
-import org.gradle.api.Project
-import org.gradle.kotlin.dsl.container
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Nested
+import org.gradle.kotlin.dsl.domainObjectContainer
 import org.gradle.kotlin.dsl.newInstance
-import org.gradle.kotlin.dsl.property
 import java.time.Duration
+import javax.inject.Inject
 
-open class NexusPublishExtension(project: Project) {
+abstract class NexusPublishExtension @Inject constructor(objects: ObjectFactory) {
```

Comment:
Hey @3flex! I did a similar migration as a try to fix an issue, check out this branch:
https://github.com/TWiStErRob/nexus-publish-gradle-plugin/commits/configuration-cache-induced-idiomatic-gradle-improvements

You might be able to cherry-pick or merge changes from there to complete some not needed constructor-injections.

I'm happy to rebase it on top of this branch too and raise a PR to yours, if you don't want to mess about.

Reply:
That's great if you're working on similar changes, I'll leave this one as is, then you can continue on your branch?